contrib/examples/actions/workflows/customfilters/mistral-regex_match.yaml

```diff
@@ -0,0 +1,20 @@
+version: '2.0'
+
+examples.mistral-customfilters-regex_match:
+    description: A workflow for testing regex_match custom filter in mistral
+    type: direct
+    vars:
+        regex_pattern: "([0-9]{3} \\w+ (?:Ave|St|Dr))"
+    input:
+        - input_str
+    output:
+        result_jinja: <% $.result_jinja %>
+        result_yaql: <% $.result_yaql %>
+    tasks:
+
+        task1:
+            action: std.noop
+            publish:
+                result_jinja: '{{ _.input_str | regex_match(_.regex_pattern)}}'
```

Comment:
So I was wrong, there is a technical reason to use this notation (at least the way the underlying function is implemented of course).  Using these functions in Jinja in the same way that they're used in YAQL will pass context as the first argument. So when I made this change, I got a lot of these types of errors:

```
JinjaEvaluationException: Can not evaluate Jinja expression [expression= version_strip_patch(_.version) , error=version_strip_patch() takes exactly 1 argument (2 given), data={}]
```

None of these filters actually require direct access to context, only the value passed in, so while it would be possible to modify the underlying implementation to access context, I'm leaning towards leaving things the way they are because 1) least-privilege principles and 2) the filter format (`|`) is the generally expected way this is done. Thoughts?